fbw-a380x/src/wasm/systems/a380_systems/src/fuel/fuel_quantity_management_system.rs

```diff
@@ -309,49 +309,44 @@ impl RefuelApplication {
 
     fn lookup_trim_fuel_from_target_fuel_range(
         &mut self,
-        category_index: usize,
+        zfw_category: &HashMap<u32, Vec<u32>>,
         total_fuel_desired_rounded: u32,
         zero_fuel_weight_cg_rounded: u32,
     ) -> Mass {
-        let zfw_category = &self.trim_tank_map.trim_tank_targets.tables[category_index];
         let target_fuel_range = zfw_category.get(&total_fuel_desired_rounded).unwrap();
 
-        if let Some(cg_index) = self
+        let cg_index = self
             .trim_tank_map
             .params
             .target_zfw_cg
             .iter()
             .position(|&x| x == zero_fuel_weight_cg_rounded)
-        {
-            return Mass::new::<kilogram>(target_fuel_range[cg_index] as f64);
-        } else {
-            return Mass::new::<kilogram>(target_fuel_range[0] as f64);
-        }
+            .unwrap_or_default();
+        Mass::new::<kilogram>(target_fuel_range[cg_index] as f64)
     }
 
     fn lookup_trim_fuel_from_mapping(
         &mut self,
-        category_index: usize,
+        zfw_category: &HashMap<u32, Vec<u32>>,
         total_desired_fuel_rounded: u32,
         zero_fuel_weight_cg_rounded: u32,
     ) -> Mass {
```

Comment:
```suggestion
    fn lookup_trim_fuel_from_mapping(
        target_zfw_cg: &[u32],
        zfw_category: &HashMap<u32, Vec<u32>>,
        total_desired_fuel_rounded: u32,
        zero_fuel_weight_cg_rounded: u32,
    ) -> Mass {
```
Same as above: pass `target_zfw_cg` as parameter and make function static. Pass `target_zfw_cg` to `Self::lookup_trim_fuel_from_target_fuel_range(...)`